tests/test_kernels.py

```diff
@@ -76,105 +72,75 @@ def test_required_grid_vars(grid_type_field_and_extra_kwargs):
-        delta = np.zeros_like(data)
-        ny = np.shape(delta)[0]
-        random_yloc = np.random.randint(5, ny - 2)
-        nx = np.shape(delta)[1]
-        random_xloc = np.random.randint(0, nx)
-        delta[random_yloc, random_xloc] = 1
-
-        test_kwargs = copy.deepcopy(extra_kwargs)
-        ones = np.ones_like(data)
-        # create highly irregular grid data for
-        # - southern edge of cell {j-1,i};
-        # - southern edge of cell {j+2,i};
-        # this should leave laplacian(delta_{j,i}) unaffected
-        if grid_type == GridType.IRREGULAR_WITH_LAND:
-            test_kwargs["wet_mask"] = ones  # no land for simplicity
-            test_kwargs["area"] = ones
-            test_kwargs["dyw"] = ones
-            test_kwargs["dys"] = ones
-            test_kwargs["dxw"] = ones
-            test_kwargs["dxs"] = ones.copy()
-            test_kwargs["dxs"][random_yloc - 1, :] = 1000
-            test_kwargs["dxs"][random_yloc + 2, :] = 2000
-        if grid_type == GridType.TRIPOLAR_POP_WITH_LAND:
-            test_kwargs["wet_mask"] = ones.copy()
-            test_kwargs["wet_mask"][0, :] = 0  # Antarctica
-            test_kwargs["tarea"] = ones
-            test_kwargs["dye"] = ones
-            test_kwargs["dyn"] = ones
-            test_kwargs["dxe"] = ones
-            test_kwargs["dxn"] = ones.copy()
-            test_kwargs["dxn"][random_yloc - 2, :] = 1000
-            test_kwargs["dxn"][random_yloc + 1, :] = 2000
+    if grid_type not in irregular_grids:
+        pytest.skip("This test is only for irregular grids")
+
+    # deploy mass at random location away from Antarctica: delta_{j,i}
+
+    delta = np.zeros_like(data)
+    ny, nx = delta.shape
+    # pick a location outside of the mask
+    random_yloc = 99
+    random_xloc = 225
+    delta[random_yloc, random_xloc] = 1
+
+    test_kwargs = extra_kwargs.copy()
+
+    replace_data = {
+        GridType.IRREGULAR_WITH_LAND: {
+            "Y": (
+                "dxs",
+                (random_yloc - 1, slice(None)),
+                (random_yloc + 2, slice(None)),
+            ),
+            "X": (
+                "dyw",
+                (slice(None), random_xloc - 1),
+                (slice(None), random_xloc + 2),
+            ),
+        },
+        GridType.TRIPOLAR_POP_WITH_LAND: {
+            "Y": (
+                "dxn",
+                (random_yloc - 2, slice(None)),
+                (random_yloc + 1, slice(None)),
+            ),
+            "X": (
+                "dye",
+                (slice(None), random_xloc - 2),
+                (slice(None), random_xloc + 1),
+            ),
+        },
+    }
+
+    var_to_modify, slice_left, slice_right = replace_data[grid_type][direction]
+    print(grid_type, var_to_modify, slice_left, slice_right)
+    new_data = np.ones_like(test_kwargs[var_to_modify])
+    new_data[slice_left] = 1000
+    new_data[slice_right] = 2000
+    test_kwargs[var_to_modify] = new_data
```

Comment:
@NoraLoose - I did a pretty big refactoring of these "flux" tests. I'd like to make sure you approve these changes and understand how the new code works.

Reply:
@rabernat, this looks great to me. Thanks for cleaning up the code!